### PR TITLE
fix windows WAL lock

### LINKED ISSUE #1972: windows: fix WAL lock
@chai2010 said there is a windows file locker (not include TryLock):
https://code.google.com/p/leveldb-go/source/browse/leveldb/db/file_lock_windows.go

We need someone to write, test and maintain the lock code for Windows. Right now it is a no-op in pkg/fileutil/lock_windows.go.

/cc @AndyPook @rwindelz @chai2010 @fceller

### PR DESCRIPTION
**Pull request by AdoHe**
Implement file lock on windows.

Fix #1972

### CODE REVIEW

File: pkg/fileutil/lock_windows.go [hunk 1]
```diff
@@ -49,12 +50,23 @@ func (l *lock) TryLock() error {
 
 // Lock acquires exclusivity on the lock without blocking
 func (l *lock) Lock() error {
+	p, err := syscall.UTF16PtrFromString(l.file.Name())
+	if err != nil {
+		return err
+	}
+	_, err = syscall.CreateFile(p, syscall.GENERIC_READ|syscall.GENERIC_WRITE,
```
Comment (xiang90) [inline pkg/fileutil/lock_windows.go]: we want to lock an existing file, we should not create a new file :(

File: pkg/fileutil/lock_windows.go [hunk 2]
```diff
@@ -49,12 +50,23 @@ func (l *lock) TryLock() error {
 
 // Lock acquires exclusivity on the lock without blocking
 func (l *lock) Lock() error {
+	p, err := syscall.UTF16PtrFromString(l.file.Name())
```
Comment (xiang90) [inline pkg/fileutil/lock_windows.go]: also we have to implement try lock.

Comment (AdoHe) [inline pkg/fileutil/lock_windows.go]: @xiang90 Ok, I will try to do this.

**Comment by gyuho** [comment]
@AdoHe I am closing this in favor of https://github.com/coreos/etcd/pull/4871.

Thanks!

**Pull request closed without merging**